go/consensus/tendermint/abci/mux.go

```diff
@@ -1000,11 +1000,31 @@ func (s *ApplicationState) GetBaseEpoch() (epochtime.EpochTime, error) {
 	return s.timeSource.GetBaseEpoch(s.ctx)
 }
 
-// GetEpoch returns current epoch at block height.
+// GetEpoch returns epoch at block height.
 func (s *ApplicationState) GetEpoch(ctx context.Context, blockHeight int64) (epochtime.EpochTime, error) {
 	return s.timeSource.GetEpoch(ctx, blockHeight)
 }
 
+// GetCurrentEpoch returns the epoch at the current block height,
+// applying all the weird legacy off-by-one rules we have.
+func (s *ApplicationState) GetCurrentEpoch(ctx context.Context) (epochtime.EpochTime, error) {
+	blockHeight := s.BlockHeight()
+	if blockHeight == 0 {
+		return epochtime.EpochInvalid, nil
+	} else if blockHeight == 1 {
+		currentEpoch, err := s.timeSource.GetEpoch(ctx, blockHeight)
```

Comment:
Would something break if we make this consistent? Was this because of the deferred InitChain that we used to have?

Reply:
https://github.com/oasislabs/oasis-core/commit/6b4965e1489ca37574ec999f7e3a8243ffe8aee5#diff-fd9bee7572e091f358b540058316ab3eR761 the `+1` stuff was added to fix something about queries

Reply:
testing in #2634